dpnp/dpnp_iface_sorting.py

```diff
@@ -190,18 +256,33 @@ def sort(x1, **kwargs):
     Examples
     --------
     >>> import dpnp as np
-    >>> a = np.array([1, 4, 3, 1])
-    >>> out = np.sort(a)
-    >>> [i for i in out]
-    [1, 1, 3, 4]
+    >>> a = np.array([[1,4],[3,1]])
+    >>> np.sort(a)                # sort along the last axis
+    array([[1, 4],
+           [1, 3]])
+    >>> np.sort(a, axis=None)     # sort the flattened array
+    array([1, 1, 3, 4])
+    >>> np.sort(a, axis=0)        # sort along the first axis
+    array([[1, 1],
+           [3, 4]])
 
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc and not kwargs:
-        if x1_desc.ndim != 1:
-            pass
-        else:
-            return dpnp_sort(x1_desc).get_pyobj()
-
-    return call_origin(numpy.sort, x1, **kwargs)
+    if kind is not None:
+        pass
+    elif order is not None:
+        pass
+    else:
+        dpnp.check_supported_arrays_type(a)
+        if axis is None:
+            a = a.flatten()
+            axis = -1
+        if a.ndim == 0:
+            raise numpy.AxisError(
+                f"axis {axis} is out of bounds for array of dimension {a.ndim}."
+            )
+        return dpnp_array._create_from_usm_ndarray(
+            dpt.sort(dpnp.get_usm_ndarray(a), axis=axis)
```

Comment:
This behavior is by design in `dpctl`. Do we need to align the behavior with `NumPy` in `dpnp`?

Reply:
It would cause an extra copy and so unnecessary performance drawback, if to implement this in dpnp. I hoped it might be done in dpctl, but since there was a reason to implement that this way in dpctl, I don't think we need to change something in dpnp also and so to keep the same logic as dpctl has.